File: BW/11월1주차/k진수에서소수개수구하기.js
function isPrime(num) {
	if(num <= 1) return false; 
    
 	for(var i = 2; i <= parseInt(Math.sqrt(num)); i++){
  		if(num % i === 0) return false;
	} 
    return true; 
}

function solution(n, k) {
    var answer = 0;
    var stack = n.toString(k).toUpperCase().split("");
    var newStack = "";
    
    while(stack.length > 0){
        var current = stack.shift();
        
        if(current === "0"){ // 0을 만난 경우
            if (newStack !== "" && isPrime(Number(newStack))) {
                answer++;
            }
            newStack = "";
        } else{
            newStack += current;
        }      
    }
    
    // 마지막 값 확인
    if (newStack !== "" && isPrime(Number(newStack))) {
        answer++;
    }

    return answer;
}

File: MK/10월 3주차/k진수에서 소수 개수 구하기.py
def solution(n, k):
    def trans(n,k):
        val=""
        while n:
            n, a = n // k, n%k
            val = str(a) + val
        return val

    nums = list(map(int ,list(num for num in trans(n,k).split("0") if num!="")))

    res=len(nums)

    for num in nums:
        if num in [0,1]:
            res-=1
            continue
        for i in range(2, int(num**0.5)+1):
            if num % i ==0:
                res-=1
                break

    return res